monkey/infection_monkey/post_breach/post_breach_handler.py

```diff
@@ -24,12 +25,8 @@ def execute_all_configured(self):
         """
         Executes all post breach actions.
         """
-        for pba in self.pba_list:
-            try:
-                LOG.debug("Executing PBA: '{}'".format(pba.name))
-                pba.run()
-            except Exception as e:
-                LOG.error("PBA {} failed. Error info: {}".format(pba.name, e))
+        pool = Pool(5)
+        pool.map(self.run_pba, self.pba_list)
```

Comment:
Have you tried this with multithreading? Performance was worse? Processes take longer to spawn, but are truly "parallel". If I/O is long, maybe we don't really need parallel execution, maybe we just need to init all I/O asap?

Reply:
I'm importing `Pool` from [multiprocessing.dummy](https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing.dummy).
> multiprocessing.dummy replicates the API of multiprocessing but is no more than a wrapper around the threading module.

Reply:
Ahh, I see!